frontend/packages/dev-console/src/utils/pipeline-actions.tsx

```diff
@@ -138,25 +139,33 @@ export const startPipeline: KebabAction = (
   },
 });
 
-export const rerunPipeline: KebabAction = (
-  kind: K8sKind,
-  pipeline: Pipeline,
-  latestRun: PipelineRun,
-  onSubmit?: (pipelineRun: PipelineRun) => void,
-) => {
+export const rerunPipeline: KebabAction = (kind: K8sKind, pipelineRun: PipelineRun) => {
```

Comment:
Why do we need to change this function?
Shouldn't we work with the rerupipelinerun function only

Reply:
This function used to call registered handler after creating new run. I see only one handler registered as of now & I needed the same handler for fixing this defect as well. So instead of passing the pipleline-action handler from detail pages, i've hardcoded it here. May be we can expose the handler later when we really needed it.

Reply:
A test is invoking this function with [3 arguments](https://github.com/openshift/console/blob/master/frontend/packages/dev-console/src/utils/__tests__/pipeline-actions.spec.ts#L35), please update it.

Reply:
Updated tests

Reply:
@abhi-kn - @abhinandan13jan was right - this is odd. I just noticed this now when addressing an issue with someone using this function. This function is designed to rerun pipelines, not rerun pipeline RUNS... this API isn't pure anymore. We now effectively have two re-run pipelineRuns... this will be nothing but confusing moving forward. I'm going to log an issue to address this.

Reply:
Ticket: https://jira.coreos.com/browse/ODC-2184